platforms/core-execution/persistent-cache/src/main/java/org/gradle/cache/internal/btree/BlockStore.java

```diff
@@ -40,7 +40,7 @@ public interface BlockStore {
      * Reads the first block from this store.
      */
     <T extends BlockPayload> T readFirst(Class<T> payloadType);
-    
```

Comment:
ether check or the config is broken, as check is kind of obsolete serving as a second gate, automation like spot or rewrite and handle this much quicker then investing dev resources in this kind of trivial task.